### PR TITLE
Improve module description for std::f32 and std::f64

### LINKED ISSUE #29353: API Docs: std::floating point stuff
Part of https://github.com/rust-lang/rust/issues/29329

like http://doc.rust-lang.org/std/f32/ pretty sure these are generated too, for all of them.

Here's what's needed to close out this issue:

- [ ] [the module docs](https://doc.rust-lang.org/std/f32/) should have a bit more about why this module exists.

### PR DESCRIPTION
**Pull request by palango**
Fixes #29353, see discussion there.

### CODE REVIEW

**Comment by BurntSushi** [comment]
@palango Thanks!

I'm not quite sure the docs here are enough to satisfy #29329.

It might be worth saying more about the distinction of constants defined in `std::{f32,f64}` and `std::{f32,f64}::consts`. For example, something like, "constants in this module are specific to the floating point implementation while constants in the `consts` sub-module contain mathematically significant numbers."

cc @steveklabnik @rust-lang/libs

File: src/libstd/f32.rs [hunk 1]
```diff
@@ -8,7 +8,10 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! The 32-bit floating point type.
+//! This module provides access to properties and bounds of the 32-bit
+//! floating point data type.
+//! Mathematical constants are provided in the
+//! [`consts`](../f32/consts) module.
```
Comment (frewsxcv) [inline src/libstd/f32.rs]: I _think_ this URL should just be:

```markdown
...provided in the [`consts`][consts] module.
```

or better yet:


```markdown
...provided in the [`consts`] module.

[`consts`]: consts
```

Comment (palango) [inline src/libstd/f32.rs]: Thanks for the hint.

**Comment by palango** [comment]
Ok, I need some tips for getting the links to work.

If the link is done with `[consts]: consts` then I get the following error:
```
[00:43:58] std/f32/index.html:54: directory link - std/f32/consts
[00:43:58] std/index.html:452: broken link - std/consts
[00:43:58] std/index.html:462: broken link - std/consts
[00:44:00] std/f64/index.html:54: directory link - std/f64/consts
```

If I use `[`consts`]: {f32|f64}/consts` I get the following output, but one of of both always seems to be wrong.
```
[00:42:03] std/f32/index.html:54: broken link - std/f32/f32/consts
[00:42:03] std/index.html:452: directory link - std/f32/consts
[00:42:03] std/index.html:462: directory link - std/f64/consts
[00:42:05] std/f64/index.html:54: broken link - std/f64/f64/consts
```

**Comment by frewsxcv** [comment]
Ah, this is an example where linking doesn't always work. It looks like these links get rendered on different pages that live at different levels of the URL hierarchy. We're working on fixing this and making this easier in the future. Sorry for the confusion here, you can just remove the links for now @palango. Thanks for being patient! :)

**Comment by frewsxcv** [comment]
Thanks! @bors r+ rollup

**Pull request merged**